### PR TITLE
Add LambdaModule container

### PR DESCRIPTION
**Pull request by roulbac**
This module can be very useful if one wants to use some pytorch functions and wrap them in modules.
For example, build a lambda with torch.cat or torch.add to implement skip connections; or define a custom loss function using a lambda and other pytorch functions

### CODE REVIEW

**Review by apaszke — COMMENTED** [review]
It's not a container for submodules, so it shouldn't be in `container.py`. Also, let's rename it to `FunctionModule`, because it doesn't only work with lambda functions

**Comment by vadimkantorov** [comment]
A suggestion: `**kwargs` argument for the constructor of `LambdaContainer`. Then these arguments can be passed to the passed function. Many PyTorch standard functions can then be wrapped, like `interpolate` and others, and parameters such as `interpolate`'s `mode` and others can be passed.

Another suggestion: Sequential's `add` may wrap the passed argument in `LambdaContainer` if the passed argument isn't `nn.Module` object. then we can write things like:
```
mlp = nn.Sequential()
mlp.add(...)
mlp.add(FF.interpolate, mode = 'bilinear')
```

**Comment by Roulbac** [comment]
@apaszke I agree, this can be generalized to any callable

**Comment by apaszke** [comment]
Yeah I think making this like `functools.partial` seems like a nice idea too.

**Comment by vadimkantorov** [comment]
Another suggestion: `__repr__` should print the additional kwargs and the name of the inner function (if a known non-lambda function is passed), if it's possible

**Comment by vadimkantorov** [comment]
A whole other question is how this should serialize or even be loaded in C++ frontend

**Comment by Roulbac** [comment]
I just added it in a new functionmodule.py file in nn.modules

**Comment by ezyang** [comment]
I suppose that it could be made serializable if the passed in function was annotated with torch script.

We're going to need some tests for this code before we can merge it.
